src/regdesk/states/new_search_state.js

```diff
@@ -0,0 +1,65 @@
+import { RegState } from './reg_state.js';
+
+/**
+ * New Search state
+ */
+export class NewSearchState extends RegState {
+  /**
+   * Get the transition events this state can process.
+   */
+  static get events() {
+    // TODO: This is apparently how TS does enums, make it an enum someday?
+    return {
+      START_SEARCH: 'START_SEARCH',
+      RESET: 'RESET',
+    };
+  }
+
+  /**
+   * Initializes a new instance of the NewSearch class.
+   * @param {RegMachineArgs} regMachineArgs - Standard argument set for states.
+   * @param {*} eventMap - Mapping of this object's transition events to new states.
+   */
+  constructor(regMachineArgs, eventMap) {
+    super(regMachineArgs, eventMap);
+  }
+
+  /**
+   * Enter this state.
+   */
+  enterState() {
+    this.leftButton.hide();
+    this.rightButton.hide();
+
+    this.centerField.innerHTML = this.#getCenterTemplate();
+    this.centerField.querySelector('#newSearchText').focus();
```

Comment:
Feels kind of weird to have the RegState handling most of the dom retrieval (eg querying for left/right button) but then this class has to also do dom retrieval. Can they be centralized?

Reply:
This specific line was a bit of an afterthought to see how hard it would be to ensure that when the page arrives it's ready to be typed into. 

That said, I consider the page-state-object to be _scoped_ to controlling the center field while it's entered the state. The enter method should be doing whatever is necessary to make that center field ready to go for the state's behavior which may include performing DOM queries within the field it's controlling. I don't think centralizing the painting of the center field is going to work well when each state object is going to be doing its own thing with it.

There's a split here between painting the page and handling the behaviors of the page that, given more time for a more robust architecture, I'd prefer to build. I just don't know that I have time..

Reply:
Okay in response to the three related comments, each state manages its own dedicated _row_ instead of the entire center element. The row IDs are derived directly off the class name automatically and are the only thing the RegState class exposes to the derived classes.

This moves the HTML over to just the HTML and we limit selector operations to just the class constructors.

Individual state classes still need to do their own dom queries unless we want to have one giant map object which I'm adamantly against based on previous horror stories involving autogenerated object maps for automated web GUI testing. I think scoped queries are okay here as we've given the state classes a sandbox within which they're meant to play.